docusaurus/docs/create-a-plugin/develop-a-plugin/good-practices.md

```diff
@@ -0,0 +1,66 @@
+---
+id: good-practices
+title: Good practices
+description: An index of good practices for plugins.
+keywords:
+  - grafana
+  - plugins
+  - plugin
+  - good practices
+sidebar_position: 5
+---
+
+This document summarizes good practices for Grafana plugins. This live document is updated as new good practices are identified.
+
+## General
+
+- Choose the right plugin type for your use case: panel, data-source or app. See [Plugin types](https://grafana.com/developers/plugin-tools/introduction/plugin-types-usage) for more information.
+- Verify that your data-source or app plugin can be provisioned. See [Provisioning](https://grafana.com/docs/grafana/latest/administration/provisioning/#data-sources) for more information.
+- Include default dashboards in your data-source or app plugin. See [Bundling of dashboards](https://grafana.com/docs/grafana/latest/administration/provisioning/#data-sources) for more information.
+- Ensure that the minimum version of Grafana is correct. This should point to the earliest version of Grafana that your plugin fully supports. This is specified as `grafanaDependency` in your plugin.json.
+
+## Panel plugins
+
+- Do not store or use credentials. Panel plugins do not have a way to securely store credentials. If your plugin needs to use credentials, consider using a data-source plugin instead and a panel plugin to display the information returned by the data-source.
+- See how you can add more advanced options to your panel plugin [here](https://grafana.com/developers/plugin-tools/create-a-plugin/extend-a-plugin/custom-panel-option-editors).
+
+## Data-source plugins
+
+- If your plugin needs to store credentials, use `secureJsonData` instead of `jsonData`. The former is encrypted at rest while the latter is not. See [Secure JSON data](https://grafana.com/developers/plugin-tools/create-a-plugin/extend-a-plugin/add-authentication-for-data-source-plugins#store-configuration-in-securejsondata) for more information.
```

Comment:
```suggestion
- If your plugin needs to store credentials, use `secureJsonData` instead of `jsonData`. The former is encrypted at rest while the latter is not. See [Secure JSON data](/create-a-plugin/extend-a-plugin/add-authentication-for-data-source-plugins#store-configuration-in-securejsondata) for more information.
```